internal/mempool/mempool.go

```diff
@@ -872,19 +883,228 @@ func (g *txDescGraph) forEachDescendant(txHash *chainhash.Hash,
-		descendantStats := mv.ancestorStats[*descendant.Tx.Hash()]
-		descendantStats.Fees -= txDesc.Fee
-		descendantStats.SizeBytes -= txSize
-		descendantStats.TotalSigOps -= txDesc.TotalSigOps
-		descendantStats.NumAncestors--
-	})
+func (mv *txMiningView) updateStatsDescendantsRemoved(baseTxDesc *mining.TxDesc) {
+	baseTxHash := baseTxDesc.Tx.Hash()
+	if _, hasStats := mv.ancestorStats[*baseTxHash]; !hasStats {
+		// If the transaction does not have ancestor tracking enabled, then
+		// none of its descendants should either.
+		return
+	}
+
+	numUntrackedDescendants := 0
+	seenDescendants := make(map[chainhash.Hash]struct{}, ancestorTrackingLimit)
+	mv.txGraph.forEachDescendantPreOrder(baseTxHash, seenDescendants,
+		func(descendantTxDesc *mining.TxDesc) bool {
+			descendantTxHash := *descendantTxDesc.Tx.Hash()
+			descendantStats, hasStats := mv.ancestorStats[descendantTxHash]
+
+			// Attempt to track the descendant's ancestor stats if it was not
+			// tracked previously. This scenario applies to a transaction
+			// that once had too many ancestors, but is now eligible to
+			// have stats tracked.
+			// A check is also performed to avoid testing too many descendants
+			// for eligibility to be included in the tracked set, since no
+			// descendant can know if any one of its ancestors has too many
+			// descendants tracked without visiting the ancestor. Note that
+			// although this has a non-ideal behavior of potentially skipping
+			// transactions that would otherwise be eligible to have stats
+			// tracked, it bounds the number of times ancestors are walked.
+			if !hasStats && numUntrackedDescendants < ancestorTrackingLimit {
+				numUntrackedDescendants++
+
+				// Note that this call retrieves ancestors while walking
+				// descendants of another transaction, and that the time
+				// complexity is bound by the maximum number of ancestors
+				// allowed to be tracked in the graph and the maximum
+				// number of descendants allowed to have ancestor stats tracked.
+				mv.maybeUpdateAncestorStats(descendantTxDesc, baseTxHash)
+
+				// Do not walk descendants of this descendant if it did not
+				// have ancestor stats previously because it was on the edge of
+				// the ancestor limit. Any transactions that descend from it
+				// would exceed the limit by at least one.
+				return false
+			}
+
+			if hasStats {
+				removeAncestorFrom(descendantStats, baseTxDesc)
+				return true
+			}
+
+			// If the transaction does not have statistics tracked then none of
+			// its descendants will either.
+			return false
+		})
+
+	// Update all ancestors to account for the removal of a descendant.
+	seenAncestors := make(map[chainhash.Hash]*mining.TxDesc,
+		ancestorTrackingLimit)
+	mv.txGraph.forEachAncestorPostOrder(baseTxHash, seenAncestors,
+		func(ancestorTxDesc *mining.TxDesc) bool {
+			ancestorStats, exists := mv.ancestorStats[*ancestorTxDesc.Tx.Hash()]
+			if exists {
+				removeDescendantFrom(ancestorStats, ancestorTxDesc)
```

Comment:
It seems that `ancestorTxDesc` would not be the correct transaction to pass here.  It's not impacting anything since the `txDesc` parameter is not being used in `removeDescendantFrom` or `addDescendantTo`.  That parameter should probably just be removed from those functions since it is unused.

Reply:
Removed the unused parameter from both functions, and also updated their comments. Thanks.